File: app/src/main/java/org/scp/gymlog/room/daos/ExerciseMuscleCrossRefDao.java
package org.scp.gymlog.room.daos;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;

import org.scp.gymlog.room.entities.ExerciseMuscleCrossRef;
import org.scp.gymlog.room.entities.SecondaryExerciseMuscleCrossRef;

@Dao
public interface ExerciseMuscleCrossRefDao {
    @Insert
    long insert(ExerciseMuscleCrossRef exercise);

    @Insert
    long[] insertAll(ExerciseMuscleCrossRef... exercises);

    @Delete
    void delete(ExerciseMuscleCrossRef exercise);

    @Query("DELETE FROM exercise_x_muscle_group WHERE exerciseId = :exerciseId")
    int clearMusclesFromExercise(int exerciseId);

    @Insert
    long insert(SecondaryExerciseMuscleCrossRef exercise);

    @Insert
    long[] insertAll(SecondaryExerciseMuscleCrossRef... exercises);

    @Delete
    void delete(SecondaryExerciseMuscleCrossRef exercise);

    @Query("DELETE FROM secondary_exercise_x_muscle_group WHERE exerciseId = :exerciseId")
    int clearSecondaryMusclesFromExercise(int exerciseId);
}

File: app/src/main/java/org/oar/gymlog/room/daos/ExerciseMuscleCrossRefDao.kt
package org.oar.gymlog.room.daos

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query
import org.oar.gymlog.room.entities.ExerciseMuscleCrossRef
import org.oar.gymlog.room.entities.SecondaryExerciseMuscleCrossRef

@Dao
interface ExerciseMuscleCrossRefDao {
    @Query("SELECT * FROM exercise_x_muscle_group")
    fun getAll(): List<ExerciseMuscleCrossRef>

    @Insert
    fun insert(exercise: ExerciseMuscleCrossRef): Long

    @Insert
    fun insertAll(exercises: List<ExerciseMuscleCrossRef>): LongArray

    @Delete
    fun delete(exercise: ExerciseMuscleCrossRef)

    @Query("DELETE FROM exercise_x_muscle_group WHERE exerciseId = :exerciseId")
    fun clearMusclesFromExercise(exerciseId: Int): Int

    @Query("DELETE FROM exercise_x_muscle_group")
    fun clear()



    @Query("SELECT * FROM secondary_exercise_x_muscle_group")
    fun getAllSecondaryMuscles(): List<SecondaryExerciseMuscleCrossRef>

    @Insert
    fun insert(exercise: SecondaryExerciseMuscleCrossRef): Long

    @Insert
    fun insertAllSecondaries(exercises: List<SecondaryExerciseMuscleCrossRef>): LongArray

    @Delete
    fun delete(exercise: SecondaryExerciseMuscleCrossRef)

    @Query("DELETE FROM secondary_exercise_x_muscle_group WHERE exerciseId = :exerciseId")
    fun clearSecondaryMusclesFromExercise(exerciseId: Int): Int

    @Query("DELETE FROM secondary_exercise_x_muscle_group")
    fun clearSecondary()
}
